core/core/src/observability/mod.rs

```diff
@@ -0,0 +1,84 @@
+use tracing::metadata::LevelFilter;
+use tracing_subscriber::{
+    filter::FilterFn, fmt::format::json, layer::SubscriberExt, EnvFilter,
+    Layer, util::SubscriberInitExt,
+};
+
+use self::buffer::{SharedEventBuffer, VecEventBuffer, RingEventBuffer};
+
+pub mod buffer;
+
+static mut METRICS_BUFFER: Option<SharedEventBuffer<VecEventBuffer>> = None;
+static mut DEVELOPER_DUMP_BUFFER: Option<SharedEventBuffer<RingEventBuffer>> = None;
+
+/// SAFETY: must only be called once during initialization of the program
+pub(crate) unsafe fn init(ring_event_buffer_size: usize) {
```

Comment:
Why is it `pub(crate)`?

Reply:
It was previously taking some arguments which I didn't want to necessarily make public for clearer intention. But now that the module is self-enclosed it can be `pub` again.